src/panel_gwalker/_gwalker.py

```diff
@@ -54,28 +222,112 @@ class GraphicWalker(ReactComponent):
+    spec: SpecType = Spec(
+        doc="""Optional chart specification as url, json, dict or list.
+    Can be generated via the `export` method."""
+    )
     server_computation: bool = param.Boolean(
         default=False,
         doc="""If True the computations will take place on the Panel server or in the Jupyter kernel
         instead of the client to scale to larger datasets. Default is False. In Pyodide this will
-        always be set to False.""",
+        always be set to False. For the chart renderer computations will always be done in the client.""",
         constant=state._is_pyodide,
     )
     config: dict = param.Dict(
         doc="""Optional extra Graphic Walker configuration. For example `{"i18nLang": "ja-JP"}`. See the
     [Graphic Walker API](https://github.com/Kanaries/graphic-walker#api) for more details."""
     )
+    renderer: Literal["explorer", "profiler", "viewer", "chart"] = param.Selector(
+        default="explorer",
+        objects=["explorer", "profiler", "viewer", "chart"],
+        doc="""How to display the data. One of 'explorer' (default), 'profiler,
+        'viewer' or 'chart'.""",
+    )
+    index: int | list[int] | None = param.ClassSelector(
+        class_=(int, list, type(None)),
+        doc="""An optional chart index or list of chart indices to display in the 'viewer' or 'chart' renderer.
+    Has no effect on other renderers.""",
+    )
+    page_size: int = param.Integer(
+        20,
+        bounds=(1, None),
+        doc="""The number of rows per page in the table of the 'profiler' render.
+    Has no effect on other renderers.""",
+    )
+    tab: Literal["data", "vis"] = param.Selector(
+        default="vis",
+        objects=["data", "vis"],
+        doc="""Set the active tab to 'data' or 'vis' (default). Only applicable for the 'explorer' renderer. Not bi-directionally synced with client.""",
+    )
+    container_height: str = param.String(
+        default="400px",
+        doc="""The height of a single chart in the 'viewer' or 'chart' renderer. For example '500px' (pixels) or '30vh' (viewport height).""",
+    )
 
     appearance: Literal["media", "dark", "light"] = param.Selector(
         default="light",
         objects=["light", "dark", "media"],
         doc="""Dark mode preference: 'light', 'dark' or 'media'.
         If not provided the appearance is derived from pn.config.theme.""",
     )
-    theme: Literal["vega", "g2", "streamlit"]=param.Selector(
-        default="vega",
-        objects=["vega", "g2", "streamlit"],
-        doc="""The theme of the chart(s). One of 'vega' (default), 'g2' or 'streamlit'.""",
+    theme_key: Literal["g2", "streamlit", "vega"] = param.Selector(
+        default="g2",
+        objects=["g2", "streamlit", "vega"],
+        doc="""The theme of the chart(s). One of 'g2', 'streamlit' or 'vega' (default).""",
+    )
+
+    export_mode: Literal["spec", "svg"] = param.Selector(
```

Comment:
Would probably prefer if these were arguments to the `export_controls` method, which then sets those arguments on the `ExportControls` Viewer object.

Reply:
These are also used as default arguments for the `export` method. Thus you cannot put them on the `ExportControls` alone.

Reply:
I put them here deliberately to make things easier for users. They might want to use the `ExportControl` or combine parameters in the own way. But creating the parameters as well takes time because you have to look up which options are available. Why not keep track of this for users?